# Data exploration: GDP

- This notebook will explore the data regarding the relation between the food consumption and the GDP of the countries.


- This dataset only has food expenditure data for the following years: 2015 & 2016


- It present the following queries:  
  - GDP of the countries with the highest percentage of income spent in food  
  - The highest share expenditure countries have a high/low GDP? (in reference to the average)  
  - The lowest share expenditure countries have a high/low GDP? (in reference to the average)  

#### Connection to DB

In [10]:
import mysql.connector
%load_ext sql
%sql mysql+mysqldb://root:admin@localhost/food_stat

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


#### What GDP have the countries where the citizens spend the biggest percentage of income in food (Year 2016) (Use of Join)

In [2]:
%%sql
SELECT t1.country, t1.year, t1.share_expenditure, t2.gdp_per_capita
FROM food_stat.share_food_expenditure as t1
JOIN food_stat.annual_food_expenditure_gdp as t2
ON (t1.country=t2.country and t1.year=t2.year)
WHERE share_expenditure is NOT NULL and t1.year=2016
ORDER BY t1.share_expenditure DESC
LIMIT 10;

 * mysql+mysqldb://root:***@localhost/food_stat
10 rows affected.


[('Nigeria', 2016, 59, 2176.0),
 ('Kenya', 2016, 52, 1410.53),
 ('Cameroon', 2016, 45, 1364.33),
 ('Kazakhstan', 2016, 43, 7714.84),
 ('Pakistan', 2016, 42, 1368.43),
 ('Philippines', 2016, 42, 3073.65),
 ('Guatemala', 2016, 41, 4173.3),
 ('Azerbaijan', 2016, 39, 3880.74),
 ('Vietnam', 2016, 39, 2192.17),
 ('Turkmenistan', 2016, 38, 6387.68)]

#### Are these countries (biggest percentage of income spent on food) among the highest GDP or lowest GDPs? What percentage they spend on food?
#### Step 1: Create a table with a new column, where you clasify each country GDP as below the average global GDP (Low) or above the average global GDP (High). (Use of Join & Union)

In [12]:
%%sql
SELECT t1.country,
       t1.year,
       t1.share_expenditure,
       t2.gdp_per_capita,
       'LowerGDP_than_AVG'
FROM
  (SELECT AVG(t2.gdp_per_capita) AS med
   FROM food_stat.share_food_expenditure AS t1
   JOIN food_stat.annual_food_expenditure_gdp AS t2 ON (t1.country=t2.country
                                                        AND t1.year=t2.year)
   WHERE share_expenditure IS NOT NULL
     AND t1.year=2016
     AND t1.code IS NOT NULL) AS GDP_avg,
     food_stat.share_food_expenditure AS t1
JOIN food_stat.annual_food_expenditure_gdp AS t2 ON (t1.country=t2.country
                                                     AND t1.year=t2.year)
WHERE t2.gdp_per_capita < GDP_avg.med
  AND t1.food_expenditure IS NOT NULL
  AND t1.year=2016
  AND t1.code IS NOT NULL
UNION
SELECT t1.country,
       t1.year,
       t1.share_expenditure,
       t2.gdp_per_capita,
       'HigherGDP_than_AVG'
FROM
  (SELECT AVG(t2.gdp_per_capita) AS med
   FROM food_stat.share_food_expenditure AS t1
   JOIN food_stat.annual_food_expenditure_gdp AS t2 ON (t1.country=t2.country
                                                        AND t1.year=t2.year)
   WHERE share_expenditure IS NOT NULL
     AND t1.year=2016
     AND t1.code IS NOT NULL) AS GDP_avg,
     food_stat.share_food_expenditure AS t1
JOIN food_stat.annual_food_expenditure_gdp AS t2 ON (t1.country=t2.country
                                                     AND t1.year=t2.year)
WHERE t2.gdp_per_capita >= GDP_avg.med
  AND t1.food_expenditure IS NOT NULL
  AND t1.year=2016
  AND t1.code IS NOT NULL;

 * mysql+mysqldb://root:***@localhost/food_stat
84 rows affected.


[('Algeria', 2016, 37, 3946.45, 'LowerGDP_than_AVG'),
 ('Argentina', 2016, 28, 12790.2, 'LowerGDP_than_AVG'),
 ('Azerbaijan', 2016, 39, 3880.74, 'LowerGDP_than_AVG'),
 ('Belarus', 2016, 36, 5022.63, 'LowerGDP_than_AVG'),
 ('Bolivia', 2016, 29, 3076.66, 'LowerGDP_than_AVG'),
 ('Bosnia and Herzegovina', 2016, 31, 4994.69, 'LowerGDP_than_AVG'),
 ('Brazil', 2016, 16, 8710.1, 'LowerGDP_than_AVG'),
 ('Bulgaria', 2016, 19, 7548.85, 'LowerGDP_than_AVG'),
 ('Cameroon', 2016, 45, 1364.33, 'LowerGDP_than_AVG'),
 ('Chile', 2016, 16, 13753.6, 'LowerGDP_than_AVG'),
 ('China', 2016, 22, 8147.94, 'LowerGDP_than_AVG'),
 ('Colombia', 2016, 17, 5870.78, 'LowerGDP_than_AVG'),
 ('Costa Rica', 2016, 26, 12011.2, 'LowerGDP_than_AVG'),
 ('Croatia', 2016, 24, 12361.5, 'LowerGDP_than_AVG'),
 ('Czechia', 2016, 17, 18575.2, 'LowerGDP_than_AVG'),
 ('Dominican Republic', 2016, 27, 7280.88, 'LowerGDP_than_AVG'),
 ('Ecuador', 2016, 22, 6060.09, 'LowerGDP_than_AVG'),
 ('Egypt', 2016, 37, 3519.87, 'LowerGDP_than_AVG'),
 ('Estonia', 2016, 21, 18437.3, 'LowerGDP_than_AVG'),
 ('Georgia', 2016, 31, 4062.17, 'LowerGDP_than_AVG'),
 ('Greece', 2016, 17, 17885.4, 'LowerGDP_than_AVG'),
 ('Guatemala', 2016, 41, 4173.3, 'LowerGDP_than_AVG'),
 ('Hungary', 2016, 18, 13090.5, 'LowerGDP_than_AVG'),
 ('India', 2016, 30, 1732.55, 'LowerGDP_than_AVG'),
 ('Indonesia', 2016, 31, 3562.82, 'LowerGDP_than_AVG'),
 ('Iran', 2016, 24, 5253.43, 'LowerGDP_than_AVG'),
 ('Jordan', 2016, 30, 4175.36, 'LowerGDP_than_AVG'),
 ('Kazakhstan', 2016, 43, 7714.84, 'LowerGDP_than_AVG'),
 ('Kenya', 2016, 52, 1410.53, 'LowerGDP_than_AVG'),
 ('Latvia', 2016, 19, 14315.8, 'LowerGDP_than_AVG'),
 ('Lithuania', 2016, 23, 14998.1, 'LowerGDP_than_AVG'),
 ('Malaysia', 2016, 21, 9817.79, 'LowerGDP_than_AVG'),
 ('Mexico', 2016, 23, 8744.52, 'LowerGDP_than_AVG'),
 ('Montenegro', 2016, 18, 7033.6, 'LowerGDP_than_AVG'),
 ('Morocco', 2016, 35, 2896.72, 'LowerGDP_than_AVG'),
 ('Nigeria', 2016, 59, 2176.0, 'LowerGDP_than_AVG'),
 ('North Macedonia', 2016, 32, 5133.0, 'LowerGDP_than_AVG'),
 ('Pakistan', 2016, 42, 1368.43, 'LowerGDP_than_AVG'),
 ('Peru', 2016, 26, 6205.0, 'LowerGDP_than_AVG'),
 ('Philippines', 2016, 42, 3073.65, 'LowerGDP_than_AVG'),
 ('Poland', 2016, 17, 12447.4, 'LowerGDP_than_AVG'),
 ('Portugal', 2016, 17, 19978.4, 'LowerGDP_than_AVG'),
 ('Romania', 2016, 30, 9548.59, 'LowerGDP_than_AVG'),
 ('Russia', 2016, 30, 8704.9, 'LowerGDP_than_AVG'),
 ('Saudi Arabia', 2016, 21, 19878.8, 'LowerGDP_than_AVG'),
 ('Serbia', 2016, 24, 5765.2, 'LowerGDP_than_AVG'),
 ('Slovakia', 2016, 18, 16508.7, 'LowerGDP_than_AVG'),
 ('South Africa', 2016, 19, 5272.54, 'LowerGDP_than_AVG'),
 ('Thailand', 2016, 26, 5993.31, 'LowerGDP_than_AVG'),
 ('Tunisia', 2016, 22, 3697.94, 'LowerGDP_than_AVG'),
 ('Turkey', 2016, 21, 10894.6, 'LowerGDP_than_AVG'),
 ('Turkmenistan', 2016, 38, 6387.68, 'LowerGDP_than_AVG'),
 ('Ukraine', 2016, 38, 2187.73, 'LowerGDP_than_AVG'),
 ('Uruguay', 2016, 19, 16715.6, 'LowerGDP_than_AVG'),
 ('Uzbekistan', 2016, 30, 2567.8, 'LowerGDP_than_AVG'),
 ('Vietnam', 2016, 39, 2192.17, 'LowerGDP_than_AVG'),
 ('Australia', 2016, 10, 49971.1, 'HigherGDP_than_AVG'),
 ('Austria', 2016, 10, 45276.8, 'HigherGDP_than_AVG'),
 ('Bahrain', 2016, 13, 22608.5, 'HigherGDP_than_AVG'),
 ('Belgium', 2016, 13, 41984.1, 'HigherGDP_than_AVG'),
 ('Canada', 2016, 9, 42315.6, 'HigherGDP_than_AVG'),
 ('Denmark', 2016, 11, 54664.0, 'HigherGDP_than_AVG'),
 ('Finland', 2016, 12, 43784.3, 'HigherGDP_than_AVG'),
 ('France', 2016, 13, 37037.4, 'HigherGDP_than_AVG'),
 ('Germany', 2016, 11, 42107.5, 'HigherGDP_than_AVG'),
 ('Hong Kong', 2016, 14, 43731.1, 'HigherGDP_than_AVG'),
 ('Ireland', 2016, 9, 62993.7, 'HigherGDP_than_AVG'),
 ('Israel', 2016, 16, 37282.5, 'HigherGDP_than_AVG'),
 ('Italy', 2016, 14, 30939.7, 'HigherGDP_than_AVG'),
 ('Japan', 2016, 14, 38761.8, 'HigherGDP_than_AVG'),
 ('Kuwait', 2016, 18, 27653.2, 'HigherGDP_than_AVG'),
 ('Netherlands', 2016, 12, 46007.9, 'HigherGDP_than_AVG'),
 ('New Zealand', 2016, 14, 40080.5, 'HigherGDP_t

#### Are these countries (biggest percentage of income spent on food) amon the highest GDP or lowest GDPs? What percentage they spend on food?
#### Step 2: Query the requested information on this table

In [13]:
%%sql
SELECT *
FROM
  (SELECT t1.country,
          t1.year,
          t1.share_expenditure,
          t2.gdp_per_capita,
          'LowerGDP_than_AVG'
   FROM
     (SELECT AVG(t2.gdp_per_capita) AS med
      FROM food_stat.share_food_expenditure AS t1
      JOIN food_stat.annual_food_expenditure_gdp AS t2 ON (t1.country=t2.country
                                                           AND t1.year=t2.year)
      WHERE share_expenditure IS NOT NULL
        AND t1.year=2016
        AND t1.code IS NOT NULL) AS GDP_avg,
        food_stat.share_food_expenditure AS t1
   JOIN food_stat.annual_food_expenditure_gdp AS t2 ON (t1.country=t2.country
                                                        AND t1.year=t2.year)
   WHERE t2.gdp_per_capita < GDP_avg.med
     AND t1.food_expenditure IS NOT NULL
     AND t1.year=2016
     AND t1.code IS NOT NULL
   UNION SELECT t1.country,
                t1.year,
                t1.share_expenditure,
                t2.gdp_per_capita,
                'HigherGDP_than_AVG'
   FROM
     (SELECT AVG(t2.gdp_per_capita) AS med
      FROM food_stat.share_food_expenditure AS t1
      JOIN food_stat.annual_food_expenditure_gdp AS t2 ON (t1.country=t2.country
                                                           AND t1.year=t2.year)
      WHERE share_expenditure IS NOT NULL
        AND t1.year=2016
        AND t1.code IS NOT NULL) AS GDP_avg,
        food_stat.share_food_expenditure AS t1
   JOIN food_stat.annual_food_expenditure_gdp AS t2 ON (t1.country=t2.country
                                                        AND t1.year=t2.year)
   WHERE t2.gdp_per_capita >= GDP_avg.med
     AND t1.food_expenditure IS NOT NULL
     AND t1.year=2016
     AND t1.code IS NOT NULL) AS full_table
ORDER BY share_expenditure DESC
LIMIT 10;

 * mysql+mysqldb://root:***@localhost/food_stat
10 rows affected.


[('Nigeria', 2016, 59, 2176.0, 'LowerGDP_than_AVG'),
 ('Kenya', 2016, 52, 1410.53, 'LowerGDP_than_AVG'),
 ('Cameroon', 2016, 45, 1364.33, 'LowerGDP_than_AVG'),
 ('Kazakhstan', 2016, 43, 7714.84, 'LowerGDP_than_AVG'),
 ('Philippines', 2016, 42, 3073.65, 'LowerGDP_than_AVG'),
 ('Pakistan', 2016, 42, 1368.43, 'LowerGDP_than_AVG'),
 ('Guatemala', 2016, 41, 4173.3, 'LowerGDP_than_AVG'),
 ('Vietnam', 2016, 39, 2192.17, 'LowerGDP_than_AVG'),
 ('Azerbaijan', 2016, 39, 3880.74, 'LowerGDP_than_AVG'),
 ('Ukraine', 2016, 38, 2187.73, 'LowerGDP_than_AVG')]

#### The countries with the lowest percentage of food expenditure have a higher or lower than average GDP?

In [16]:
%%sql
SELECT *
FROM
  (SELECT t1.country,
          t1.year,
          t1.share_expenditure,
          t2.gdp_per_capita,
          'LowerGDP_than_AVG'
   FROM
     (SELECT AVG(t2.gdp_per_capita) AS med
      FROM food_stat.share_food_expenditure AS t1
      JOIN food_stat.annual_food_expenditure_gdp AS t2 ON (t1.country=t2.country
                                                           AND t1.year=t2.year)
      WHERE share_expenditure IS NOT NULL
        AND t1.year=2016
        AND t1.code IS NOT NULL) AS GDP_avg,
        food_stat.share_food_expenditure AS t1
   JOIN food_stat.annual_food_expenditure_gdp AS t2 ON (t1.country=t2.country
                                                        AND t1.year=t2.year)
   WHERE t2.gdp_per_capita < GDP_avg.med
     AND t1.food_expenditure IS NOT NULL
     AND t1.year=2016
     AND t1.code IS NOT NULL
   UNION SELECT t1.country,
                t1.year,
                t1.share_expenditure,
                t2.gdp_per_capita,
                'HigherGDP_than_AVG'
   FROM
     (SELECT AVG(t2.gdp_per_capita) AS med
      FROM food_stat.share_food_expenditure AS t1
      JOIN food_stat.annual_food_expenditure_gdp AS t2 ON (t1.country=t2.country
                                                           AND t1.year=t2.year)
      WHERE share_expenditure IS NOT NULL
        AND t1.year=2016
        AND t1.code IS NOT NULL) AS GDP_avg,
        food_stat.share_food_expenditure AS t1
   JOIN food_stat.annual_food_expenditure_gdp AS t2 ON (t1.country=t2.country
                                                        AND t1.year=t2.year)
   WHERE t2.gdp_per_capita >= GDP_avg.med
     AND t1.food_expenditure IS NOT NULL
     AND t1.year=2016
     AND t1.code IS NOT NULL) AS full_table
ORDER BY share_expenditure ASC
LIMIT 10;

 * mysql+mysqldb://root:***@localhost/food_stat
10 rows affected.


[('United States', 2016, 6, 58021.4, 'HigherGDP_than_AVG'),
 ('Singapore', 2016, 7, 56848.2, 'HigherGDP_than_AVG'),
 ('United Kingdom', 2016, 8, 41048.4, 'HigherGDP_than_AVG'),
 ('Switzerland', 2016, 9, 83073.3, 'HigherGDP_than_AVG'),
 ('Canada', 2016, 9, 42315.6, 'HigherGDP_than_AVG'),
 ('Ireland', 2016, 9, 62993.7, 'HigherGDP_than_AVG'),
 ('Australia', 2016, 10, 49971.1, 'HigherGDP_than_AVG'),
 ('Austria', 2016, 10, 45276.8, 'HigherGDP_than_AVG'),
 ('Denmark', 2016, 11, 54664.0, 'HigherGDP_than_AVG'),
 ('Germany', 2016, 11, 42107.5, 'HigherGDP_than_AVG')]

#### Conclusion

- To increase the scope of what we already analyze in the Food Expenditure notebook, we compared the countries that spend more or less percentage of the income in food with the GDP they have.


- The results show that from the top 10 countries that spend the highest percentage on food they all have a GDP below the world average.


- On the contrary, the top 10 countries that spend the lowest percentage on food all have a GDP higher than the world average.


- To compare we can see the highest percentage of share expenditure corresponds to Nigeria with 59% and the lowest share expenditure corresponds to the United States with 6%.